### ISSUE TITLE
Database Container is required for free table

### ISSUE THREAD

**Issue by Raanelom** [tombrothers/VfpEntityFrameworkProvider2#10]
The code first solution of VfpEntityFrameworkProvider works like a charm, except for one issue: a free (.dbf) table as 'code first' returns the error "Database Container is required". I checked the documentation, but it seems to be somewhat outdated (https://randomdevnotes.com/2012/05/06/using-vfp-entity-framework-provider-with-free-tables-using-code-first/).

![github_evidence](https://user-images.githubusercontent.com/18660378/36163469-cdfd1e1e-10e9-11e8-9d7d-aab3bd24ad16.jpg)

The table has one candidate key 'cookie', and multiple bool, int, datetime and string fields.

![afbeelding](https://user-images.githubusercontent.com/18660378/36163653-49525494-10ea-11e8-8b9c-bc256cfd37f8.png)
![afbeelding](https://user-images.githubusercontent.com/18660378/36163675-6246082e-10ea-11e8-928e-f9a8f1f5fc03.png)

Is there a (straightforward) solution to this problem?

**Comment by tombrothers** [comment]
A DBC is only required when using Code First to generate your data structure. In your case, you are working with an existing data structure so you will want to instruct EF to not generate one for you. The way I do that is I create a static constructor on my DataContext and set the database initializer to null. 

Example:
```
public class MyDataContext : DbContext {
   static MyDataContext() {
       Database.SetInitializer<MyDataContext>(null);
   }
}
```

**Comment by Raanelom** [comment]
Thanks for your quick response.

I tried adding `Database.SetInitializer(new NullDatabaseInitializer<CookiesContext>());` (the EF-6 solution) to either a static constructor, the 'default' constructor and to global.asax. However, it seems to have no effect on the message. Perhaps there is another issue here?

My CookiesContext class:

```c#
[DbConfigurationType(typeof(VfpDbConfiguration))]
public partial class CookiesContext : DbContext
{
    public CookiesContext()
        : base("name=CookiesContext")
    {
        Database.SetInitializer(new NullDatabaseInitializer<CookiesContext>());
    }

    public virtual DbSet<Sustainability_Cookies> Cookies { get; set; }
    
    public override void OnModelCreating(DbModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Sustainability_Cookies>()
                .Property(e => e.Cookie)
                .IsFixedLength()
                .IsUnicode(false);

        // etc
    }
}
```

**Comment by tombrothers** [comment]
That should have been all you needed.  I can debug the issue for you if you could send me a sample project.

**Comment by Raanelom** [comment]
Solved the issue by adding a database container anyways. Perhaps the issue was in the web.config?

```
<add name="CookiesContext" connectionString="data source=[path-to-db]\sustainability_cookies.DBF" providerName="VfpEntityFrameworkProvider2" />
```

Not sure if a DBF can be declared in the web.config.

**Comment by tombrothers** [comment]
You are correct.  The code expected a file with a DBC extension or a directory for the Free Tables.

**Comment by pcdowning** [comment]
Vote of confidence. The solution mentioned above worked for me ...
